smartsim/builders/ensemble.py

```diff
@@ -281,7 +355,13 @@ def replicas(self, value: int) -> None:
         """Set the number of replicas.
 
         :return: the number of replicas
+        :raises TypeError: replicas argument was not of type int
         """
+        if not isinstance(value, int):
+            raise TypeError("replicas argument was not of type int")
+        if value <= 0:
+            raise ValueError
```

Comment:
Can we add a description here?
```suggestion
            raise ValueError("Number of replicas must be a positive integer")
```

Reply:
added